lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -23,7 +24,11 @@ def lint_inputs(tool_xml, lint_ctx):
             continue
 
         param_type = param_attrib["type"]
-        param_name = param_attrib.get("name", param_attrib.get("argument"))
+        param_name = ToolParameter.parse_name(param)
```

Comment:
```suggestion
        param_name = param_attrib.get('name', None)
        if param_name is None:
            param_name = param_attrib['argument'].replace("-", "_")
```
should work fine and requires less internals of Galaxy, which is a good thing for the linter.

Reply:
@mvdbeek `, None)` is not needed, it's the default for `dict.get()` .

Also your version is missing the `.lstrip('-')` .

I think that my suggestion above would still be a better solution (less code duplication), given both files are part of the `tool_util` package.

Reply:
I would also agree more with @nsoranzo

Reply:
I don't think this belongs in `XmlInputSource`, this is tool parameter logic that is independent of the type of input source. At the minimum you'd also have to add it to `YamlInputSource` or `InputSource`.

Reply:
OK. How about this: We move the actual functionality to galaxy/tool_util/parser/util.py adding a new function there, e.g. `argument_to_name`. 

This could then be used in 
* tool_util/linters/inputs.py 
* tools/parameters/basic.py

Reply:
I think moving it to `InputSource`, as suggested by Marius, is better.

Reply:
Okidoki. Lets see if I got this right :)